src/Rector/AbstractDrupalCoreRector.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DrupalRector\Rector;
+
+use Drupal\Component\Utility\DeprecationHelper;
+use DrupalRector\Contract\DrupalCoreRectorInterface;
+use DrupalRector\Contract\VersionedConfigurationInterface;
+use DrupalRector\Rector\ValueObject\FunctionToStaticConfiguration;
+use PhpParser\Node;
+use PhpParser\Node\Expr\ArrowFunction;
+use Rector\Core\Contract\Rector\ConfigurableRectorInterface;
+use Rector\Core\Rector\AbstractRector;
+
+abstract class AbstractDrupalCoreRector extends AbstractRector implements ConfigurableRectorInterface
+{
+    /**
+     * @var array|\DrupalRector\Contract\VersionedConfigurationInterface[]
+     */
+    protected array $configuration = [];
+
+    public function configure(array $configuration): void {
+        foreach ($configuration as $value) {
+            if (!($value instanceof VersionedConfigurationInterface)) {
+                throw new \InvalidArgumentException(sprintf(
+                    'Each configuration item must be an instance of "%s"',
+                    VersionedConfigurationInterface::class
+                ));
+            }
+        }
+
+        $this->configuration = $configuration;
+    }
+
+    public function refactor(Node $node)
+    {
+        $drupalVersion = str_replace(['.x-dev', '-dev'], '.0', \Drupal::VERSION);
+
+        foreach ($this->configuration as $configuration) {
+            if (version_compare($drupalVersion, $configuration->getIntroducedVersion(), '<')) {
+                continue;
+            }
+
+            $result = $this->refactorWithConfiguration($node, $configuration);
+
+            // Skip if no result.
+            if ($result === NULL) {
+                continue;
+            }
+
+            // Check if Drupal version and the introduced version support backward
+            // compatible calls. Although it was introduced in Drupal 10.1 we
+            // also supply these patches for changes introduced in Drupal 10.0.
+            // The reason for this is that will start supplying patches for
+            // Drupal 10 when 10.0 is already out of support. This means that
+            // we will not support running drupal-rector on Drupal 10.0.x.
+            if (version_compare($drupalVersion, '10.1.0', '<') || version_compare($configuration->getIntroducedVersion(), '10.0.0', '<')) {
+                return $result;
+            }
```

Comment:
Ah I think I get it. We're going to replace 10.0.x code with backwards compatibility introduced in 10.1.x because this will be run using versions of Drupal greater than 10.0.x (in bulk)

Also, ugh. I realize this means modules must require a minimum of 10.1.x if they use it. They cannot do a `^9.5 || ^10 || ^11` bridge. Oh well

Reply:
Yeah we do unfortunately. So it would be:

^9.5 | ^10.1 | ^11

But that's not for local code perse, I'll fix updatebot to do that though